### PR TITLE
feat: add LBP plugin

### PR DESCRIPTION
**Pull request by guanbinrui**
## Introduction

Lunch a Liquidity Bootstrapping Pool on Twitter.

### `<PoolDeckView />`

Presents the liquidity bootstrapping pool UI.

### Search Result View

test

### Popper View

test

## Learn More

+ https://medium.com/balancer-protocol/building-liquidity-into-token-distribution-a49d4286e0d4
+ PERP: https://perp.fi/lbp/
+ Figma: https://www.figma.com/file/gVkQ67y285b4FXVV1KPThN/Twitter?node-id=4398:212041
+ Live Demo: https://twitter.com/search?q=%23masklbp&src=typed_query
+ Kovan Testnet LBP: https://kovan.pools.balancer.exchange/#/pool/new
+ Kovan Testnet BAT against USDC pool: https://kovan.pools.balancer.exchange/#/pool/0x7047dc42334347f99971f856e12b135d39e2ef73/

### CODE REVIEW

**Comment by Jack-Works** [comment]
those abbreviations (LBP) look like a mystery to me 👀

**Comment by guanbinrui** [comment]
> those abbreviations (LBP) look like a mystery to me 👀

It's Liquidity Bootstrapping Pool as described [here](https://docs.balancer.finance/smart-contracts/smart-pools/liquidity-bootstrapping-faq).

File: packages/maskbook/package.json [hunk 1]
```diff
@@ -25,7 +25,7 @@
     "@servie/events": "^1.0.0",
     "@types/bn.js": "^4.11.6",
     "@types/classnames": "^2.2.11",
-    "@types/d3": "^6.2.0",
+    "@types/d3": "5.16.4",
```
Comment (septs) [inline packages/maskbook/package.json:28]: why need downgrade?

File: packages/maskbook/src/components/InjectedComponents/SearchResultBox.tsx [hunk 2]
```diff
@@ -1,41 +1,23 @@
...
+                    <PluginSearchResultBoxForEach config={x} />
+                </ErrorBoundary>
+            ))}
+        </>
     )
 }
+
+function PluginSearchResultBoxForEach({ config }: { config: PluginConfig }) {
+    const F = config.SearchBoxComponent
+    if (typeof F === 'function') return <F />
```
Comment (septs) [inline packages/maskbook/src/components/InjectedComponents/SearchResultBox.tsx:21]: https://npm.im/react-is

File: packages/maskbook/package.json (hunk 1 above)
Comment (guanbinrui) [reply packages/maskbook/package.json:28]: We use 5.x actually.

File: packages/maskbook/src/components/InjectedComponents/SearchResultBox.tsx (hunk 2 above)
Comment (guanbinrui) [reply packages/maskbook/src/components/InjectedComponents/SearchResultBox.tsx:21]: It's OK for this simple usage.

**Review by septs — APPROVED** [review]

**Pull request merged**